salt/states/pkg.py

```diff
@@ -2779,7 +2779,8 @@ def latest(
             failed = [
                 x
                 for x in targets
-                if not changes.get(x)
+                if changes
+                and not changes.get(x)
```

Comment:
I don't understand this change. This whole block is under a `if changes:` condition so another `if changes` should not be necessary for every item.

Reply:
Changes wasn't always defined... so it was trying to do a `.get(x)` on a NoneType.

Reply:
```
if changes:
            # Find failed and successful updates
            failed = [
                x
                for x in targets
                if changes
                and not changes.get(x)
                or changes[x].get("new") is not None
                and targets[x] not in changes[x].get("new").split(",")
                and targets[x] != "latest"
            ]
            successful = [x for x in targets if x not in failed]
```
changes has to be defined here because of the condition on line 2777 (https://github.com/saltstack/salt/blob/1ca61c5df3f880d5d4e21161c546f99c535771cd/salt/states/pkg.py#L2777C1-L2777C20)

Reply:
Also  I think it would be ideal to iterate over of targets `items()` instead of having to do a lookup again on each one

```python
for x, version in targets.items()
if not changes.get(x)
or changes[x].get("new") is not None
and version not in changes[x].get("new").split(",")
and version != "latest"
```

Reply:
Yeah, you're right. Set it back.